build.sbt

```diff
@@ -230,6 +230,7 @@ lazy val `quill-core` =
     .settings(mimaSettings: _*)
     .settings(libraryDependencies ++= Seq(
       "com.typesafe"               %  "config"        % "1.4.1",
+      "dev.zio"                    %% "zio-logging"   % "0.5.13",
```

Comment:
Can't this be done using the existing scala-logging dependency without making the core dependant on Zio? What will happen when Zio 2 is released? Will you stick to Zio 1 or release a new quill major release?
Have you considered keeping Zio in some specific modules so quill can be used without depending on it in the core?

Reply:
I've thought about going the `scala-logging`-only route. The problem is that scala-logging is just a wrapper around Slf4j and in order get the format (with Slf4j) of the file to be this:
```
line: 342
file: /my/path/to/file
pos: 34
SELECT p.name, p.age FROM Person p
```
And not this:
```
2020-03-29 11:42:44,827 INFO io.getquill.Messages [main] line: 342
2020-03-29 11:42:44,827 INFO io.getquill.Messages [main] file: /my/path/to/file
2020-03-29 11:42:44,827 INFO io.getquill.Messages [main] pos: 34
2020-03-29 11:42:44,827 INFO io.getquill.Messages [main] SELECT p.name, p.age FROM Person p
```
I basically have to import an actual logging implementaion (a.la. logback or log4j) and then configure custom appenders for them. You can't really do this kind of configuration just inside of Slf4j, since it's not an actual logger.

In terms of using ZIO in quill-core, I've been sitting on the fence for a while but I'm ready to 'take the plunge' now because there are also other parts of the codebase that could do better with ZIO stuff. For example, StatefulTransformation would be significantly more comprehensible if it used something like ZPure.

In terms of migration to ZIO 2, I'll try to do what's right for the users. If needed, I can keep a branch with ZIO 2 (or ZIO 1 for that mater) running for a while. As far as I understand, the migration will be fairly simple. I'm hoping we could even have source-compatibility across both.

Reply:
Thank you for the thorough explanation.
It makes totally sense!